data/env/Makefile

```diff
@@ -20,6 +20,14 @@ ENVD := /etc/profile.d
 	sed   < $< > $@ \
 		s:@SNAP_MOUNT_DIR@:${SNAP_MOUNT_DIR}:g
 
+# When processing snapd.sh.in remove the /bin/sh shebang as this file is meant
+# to be sourced, not executed, and the presence of that line causes rpmlint
+# validation warnings.
+snapd.sh: snapd.sh.in
```

Comment:
Can we remove it in snapd.sh directly?

Reply:
I think we can although shellcheck might complain about that. We could use ogra's advice as well but I don't know how easy that is.

Reply:
Without the shebang entirely:

```
zyga@tumbleweed:~/snapd/data/env> shellcheck snapd.sh

In snapd.sh line 1:
# Expand $PATH to include the directory where snappy applications go.
^-- SC2148: Tips depend on target shell and yours is unknown. Add a shebang.


In snapd.sh line 3:
if [ -n "${PATH##*${snap_bin_path}}" -a -n "${PATH##*${snap_bin_path}:*}" ]; then
                                     ^-- SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.


In snapd.sh line 17:
if [ -n "${XDG_DATA_DIRS##*${snap_xdg_path}}" -a -n "${XDG_DATA_DIRS##*${snap_xdg_path}:*}" ]; then
                                              ^-- SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.

```

Without the shebang but with explicit file type (not sure how to pass this elsewhere)

```
zyga@tumbleweed:~/snapd/data/env> shellcheck -s sh snapd.sh

In snapd.sh line 3:
if [ -n "${PATH##*${snap_bin_path}}" -a -n "${PATH##*${snap_bin_path}:*}" ]; then
                                     ^-- SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.


In snapd.sh line 17:
if [ -n "${XDG_DATA_DIRS##*${snap_xdg_path}}" -a -n "${XDG_DATA_DIRS##*${snap_xdg_path}:*}" ]; then
                                              ^-- SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.
```

Sadly I think this is the best we can easily do. We should also address those two issues.

Reply:
Would adding this at the top of snapd.sh work?
```
# shellcheck shell=sh
```